doc/user-guide/push-notifications/Push-notifications.md

```diff
@@ -0,0 +1,178 @@
+ ![](PushNotifications-MIM_as_remote_pubsub.png)
+
+The components that make push notifications possible in MongooseIM in the PubSub-full setup
+add up to the following architecture:
+![](component-architecture-with-PubSub.png)
+
+### MongooseIM as a PubSub-less XMPP server
+
+ ![](PushNotifications-no_pubsub.png)
+
+PubSub is completely bypassed and the clients don't need to create a push node.
+They only have to provide the virtual PubSub address.
+
+A direct connection to a push service (e.g. MongoosePush) must be configured on the same MongooseIM
+node, checkout [this][WithMongoosePush] tutorial on how to setup [MongoosePush][].
+
+If compatibility with client libraries is not an issue, this is a really good approach.
+PubSub is completely bypassed and the clients don't need to create a push node — if they attempt to
+do so, the server would respond with an error stanza, hence the incompatibility with [XEP-0357][].
+They only have to provide the virtual PubSub address.
+
+```Erlang
+{mod_event_pusher, [
+    {backends, [
+        {push, [
+            {backend, mnesia}, % optional
+            {wpool, [{workers, 200}]}, % optional
+            {plugin_module, mod_event_pusher_push_plugin_defaults}, % optional
+            {virtual_pubsub_hosts, ["virtual.@HOST@"]}
+        ]}
+    ]}
+]}
+```
+
+#### Advantages
+* No need to use PubSub at all
+* More efficient (PubSub has a considerable impact on heavily loaded systems)
+* Simpler client-side usage
+
+#### Drawbacks
+* It is only partially XEP-0357 compliant, so 3rd party clients most probably won't work even in
+  private deployments — they need to be adapted
+* Not suitable for deployments that require connections from 3rd party client applications.
+  Notifications won't work at all in such case
+
+### XEP-0357 compliant + virtual hosts
+
+This is the most versatile setup. Allows clients to use all push notifications configurations:
+
+The components that make push notifications possible in MongooseIM add up to the following architecture:
+![](component-architecture-without-PubSub.png)
+
+* PubSub-less
+* Push service handled locally via XEP-0357
+* Push service handled by another cluster via XEP-0357
+
+```Erlang
+{mod_pubsub, [{plugins, [<<"push">>]}]}, % mandatory minimal config
+
+{mod_event_pusher, [
+    {backends, [
+        {push, [
+            {backend, mnesia}, % optional
+            {wpool, [{workers, 200}]}, % optional
+            {plugin_module, mod_event_pusher_push_plugin_defaults}, % optional
+            {virtual_pubsub_hosts, ["virtual.@HOST@"]}
+        ]}
+    ]}
+]}
+```
+
+#### Virtual PubSub hosts
+
+These domains will shadow any identical domain configured for pubsub, stealing any notification
+published to it. It enables easy migration from PubSub-full deployments to PubSub-less variants.
+Read more in [mod_event_pusher_push_virtual_hosts][].
+
+#### Advantages
+* Versatility: pubsub-less and pubsub-full mechanisms can be configured with different domains and
```

Comment:
Minor detail, and I believe most files in the documentation lack consistency with this word, but in this line we have `pubsub-less` and in the line 142 there is `PubSub-less`. It just looks weird spelled differently in the same paragraph.

Reply:
Fun fact, the XEP is also mixing both spellings in the same paragraphs: https://xmpp.org/extensions/xep-0357.html 😂 
But I'll try going to the camel-case version anyway xD